src/pybind/mgr/cephadm/module.py

```diff
@@ -488,6 +501,8 @@ def serve(self) -> None:
 
             try:
 
+                self.convert_tags_to_repo_digest()
```

Comment:
nit: a comment here that the digest only changes iff the container was set to ref the image by label

having this routine in the `serve()` thread implies that the digest is always changes which is not how it's actually implemented